JASP-Engine/JASP/R/ancova.R

```diff
@@ -317,10 +321,14 @@ Ancova <- function(jaspResults, dataset = NULL, options) {
   
   model <- .anovaModel(dataset, options)
 
+  
   if (.extractErrorMessage(model$modelError) == "singular fit encountered") {
     anovaContainer$setError(gettext("Singular fit encountered; one or more predictor variables are a linear combination of other predictor variables"))
     return()
-  }
+  } else if (.extractErrorMessage(model$modelError) == "residual sum of squares is 0 (within rounding error)") {
+    anovaContainer$setError(gettext("Residual sum of squares is 0; this might be due to extremely low variance of your dependent variable"))
+    return()
+  } 
```

Comment:
What happens if an error occurs but it doesn't happen to be one of the two situations defined here?

Reply:
do you want me to add a general erroCatch with an error message of something like "An error occurred while computing the ANOVA"? I am not aware of all the possible error messages aov or afex can throw, it's just that those 2 have come up while testing (and I havent encountered other errors while testing)

Reply:
I'm not sure.. It might give more information to the user though (if we give the package error), because otherwise I'm guessing the analysis will crap out somewhere in the remainder of the ANOVA analysis and thus throw an unrelated error. Or can the analysis deal with a tryCatch result instead of an actual result?